packages/client/wallets/aa/src/blockchain/wallets/passkey.ts

```diff
@@ -0,0 +1,55 @@
+import { CrossmintWalletService, EVMAAWallet, PasskeyWalletConfig } from "@/index";
+import { getIdString } from "@/utils/user";
+import { createPasskeyValidator, deserializePasskeyValidator } from "@zerodev/passkey-validator";
+import { KernelValidator, createKernelAccount } from "@zerodev/sdk";
+import { ENTRYPOINT_ADDRESS_V07 } from "permissionless";
+import { ENTRYPOINT_ADDRESS_V07_TYPE } from "permissionless/_types/types";
+import { HttpTransport, PublicClient } from "viem";
+
+import { EVMBlockchainIncludingTestnet, UserIdentifier } from "@crossmint/common-sdk-base";
+
+export default class PasskeyWalletService {
+    private readonly entryPoint = ENTRYPOINT_ADDRESS_V07;
+    constructor(private readonly crossmintService: CrossmintWalletService, private readonly passkeyServerUrl: string) {}
+
+    public async getOrCreate(
+        userIdentifier: UserIdentifier,
+        chain: EVMBlockchainIncludingTestnet,
+        publicClient: PublicClient<HttpTransport>,
+        walletConfig?: PasskeyWalletConfig // TODO, should we let devs override the name? Or add special validation?
+    ) {
+        const serializedData = await this.get(userIdentifier, chain);
+
+        let validator: KernelValidator<ENTRYPOINT_ADDRESS_V07_TYPE, "WebAuthnValidator">;
+        if (serializedData != null) {
+            validator = await deserializePasskeyValidator(publicClient, {
+                serializedData,
+                entryPoint: this.entryPoint,
+            });
+        } else {
+            validator = await createPasskeyValidator(publicClient, {
+                passkeyServerUrl: this.passkeyServerUrl,
+                entryPoint: this.entryPoint,
+                passkeyName: getIdString(userIdentifier),
```

Comment:
we need to add the omit flag to the credentials header: https://github.com/Crossmint/crossmint-sdk/pull/485/files#diff-44d9925b3eab949d3dc6cce03789d9160cfa1d1a34305326e05e39be67f70920R77